### ISSUE TITLE
LMTP: delivery to shared mailbox broken since 3.4

### ISSUE THREAD

**Issue by mmatuska** [cyrusimap/cyrus-imapd#3488]
Configuration:
unixhierarchysep: 1
altnamespace: 1
virtdomains: userid

Maillog entry:

```
May 10 12:56:03 mail postfix/lmtp[59098]: 8BBCA1E467B: to=<+sharedmb@mydoman.com>, relay=mail.mydomain.com[/var/imap/socket/lmtp], delay=1117, delays=1117/0.03/0/0, dsn=4.3.0, status=deferred (host mail.mydomain.com[/var/imap/socket/lmtp] said: 451 4.3.0 Invalid mailbox name (in reply to end of DATA command))
```

This very same doesn't happen under 3.2.x

**Comment by swaor** [comment]
I have the same error on version 3.4.1:

`May 31 15:09:24 mail sm-mta[11475]: 14VC99rJ011475: to=+shared/mailboxname, delay=00:00:01, xdelay=00:00:00, mailer=cyrusv2, pri=1010802, relay=localhost [[UNIX: /var/imap/socket/lmtp]], dsn=4.3.0, reply=451 4.3.0 Invalid mailbox name, stat=Deferred: 451 4.3.0 Invalid mailbox name`

After downgrade to 3.2.7 - the error is gone.

Configuration is same:
unixhierarchysep: yes
altnamespace: yes
virtdomains: userid

**Comment by elliefm** [comment]
I've got a test case that works on 3.2, fails on 3.4 and master, and I've bisected it down to this being the commit that broke it: 792bc885271557e9176575f40935b57d8592e6f4.  And I can see why: the shared mailbox has no userid, so attempting to lock the user's conversations db fails.  Should have a fix up shortly.

**Comment by elliefm** [comment]
This is now fixed on the 3.4 and master branches, and will be fixed in the 3.4.2 release.

In the meantime you can fix 3.4.0 or 3.4.1 by applying this patch: https://github.com/cyrusimap/cyrus-imapd/commit/d271d002de6f9a23c449edd0b7e9dbd0a0d8d5e9.patch and rebuilding
